src/stdlib_sorting_ord_sort.fypp

```diff
@@ -105,23 +109,28 @@ contains
 ! original `listsort.txt`, and an optional `work` array to be used as
 ! scratch memory.
         ${t1}$, intent(inout)         :: array(0:)
-        ${t1}$, intent(out), optional :: work(0:)
+        ${t3}$, intent(out), optional :: work(0:)
 
-        ${t1}$, allocatable :: buf(:)
+        ${t2}$, allocatable :: buf(:)
         integer(int_size) :: array_size
         integer :: stat
 
         array_size = size( array, kind=int_size )
         if ( present(work) ) then
             if ( size( work, kind=int_size) < array_size/2 ) then
-                error stop "${k1}$_${sname}$_ord_sort: work array is too small."
+                error stop "${name1}$_${sname}$_ord_sort: work array is too small."
             endif
 ! Use the work array as scratch memory
             call merge_sort( array, work )
         else
 ! Allocate a buffer to use as scratch memory.
+            #:if t1[0] == "c"
```

Comment:
Here it might be better to check more characters (e.g.  "char" instead of "c") from a clarity perspective.

Reply:
True. One additional advantage is that more considered letters will result in less potential conflicts in the futute (e.g., in the case that we want add a new DT that starts with "c")

Reply:
Done